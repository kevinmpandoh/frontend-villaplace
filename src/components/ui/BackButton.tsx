import Link from 'next/link';

const BackButton: React.FC = () => {
  return (
    <Link href="/" passHref>
      <button 
        className="inline-flex items-center gap-2 justify-center px-3 py-2 rounded-lg hover:bg-green-100 border-2 border-primary text-primary font-semibold text-center"
        aria-label='Back'
      >
          <p>Back</p>
      </button>
    </Link>
  );
};

export default BackButton;