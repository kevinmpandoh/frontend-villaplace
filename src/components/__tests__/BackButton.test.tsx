import { render, screen } from '@testing-library/react';
import BackButton from '../ui/BackButton';
import '@testing-library/jest-dom';

jest.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: (props: any) => <svg data-testid={props['data-testid']} />,
}));

jest.mock('next/link', () => {
  return ({ children }: any) => children;
});

describe('BackButton Component', () => {
  test('Menampilkan teks "Back" dengan teks dan ikon', () => {
    render(<BackButton />);
    const backButtonText = screen.getByText('Back');
    expect(backButtonText).toBeInTheDocument();

    const icon = screen.getByTestId('fa-arrow-left');
    expect(icon).toBeInTheDocument();
  });

  test('Memiliki href yang benar dalam Link', () => {
    render(<BackButton />);
  
    const linkElement = screen.getByRole('link', { name: /back/i, hidden: true });
    expect(linkElement).toHaveAttribute('href', '/');
  });
  

  test('Snapshot testing untuk memastikan struktur elemen', () => {
    const { asFragment } = render(<BackButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
