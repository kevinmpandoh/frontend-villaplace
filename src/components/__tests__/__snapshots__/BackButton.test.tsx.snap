// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Component Snapshot testing untuk memastikan struktur elemen 1`] = `
<DocumentFragment>
  <a
    aria-label="Back"
    class="inline-flex items-center gap-2 justify-center px-3 py-2 rounded-lg hover:bg-green-100 border-2 border-primary text-primary font-semibold text-center"
  >
    <svg
      data-testid="fa-arrow-left"
    />
    <span>
      Back
    </span>
  </a>
</DocumentFragment>
`;
